src/nvim/mark.c

```diff
@@ -146,6 +146,11 @@ int setmark_pos(int c, pos_T *pos, int fnum, fmarkv_T *view_pt)
     return OK;
   }
 
+  if (c == ':' && bt_prompt(buf)) {
+    RESET_FMARK(&buf->b_prompt_start, *pos, buf->b_fnum, view);
+    return OK;
+  }
+
   if (ASCII_ISLOWER(c)) {
     i = c - 'a';
     RESET_FMARK(buf->b_namedm + i, *pos, fnum, view);
```

Comment:
side-note: would it make sense to extend `b_namedm` and drop the explicit `b_prompt_start` field? 

also just noticed something: do we need to update `NLOCALMARKS` or any other constants here?: https://github.com/neovim/neovim/blob/14e94f46177d3f562cdf9b8dc6b776fc64ca079e/src/nvim/mark_defs.h#L51-L55